<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>addTtlvNum</class>
 <widget class="QWidget" name="addTtlvNum">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>596</width>
    <height>462</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="maximumSize">
   <size>
    <width>596</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="1" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="label_8">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>高低位       </string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="comboBox_order_2">
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <item>
        <property name="text">
         <string>高8位</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>低8位</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </item>
   <item row="0" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QLabel" name="label">
       <property name="minimumSize">
        <size>
         <width>75</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>75</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>数据类型</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="comboBox_numType">
       <item>
        <property name="text">
         <string>8位有符号整形</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>8位无符号整形</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>16位有符号整形</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>16位无符号整形</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>16进制有符号AB型</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>32位有符号整形</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>32位无符号整形</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>64位有符号整形</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>64位无符号整形</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>32位单精度浮点型</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>64位双精度浮点型</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </item>
   <item row="3" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QLabel" name="label_3">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>换算公式</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_4">
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>寄存器值 * </string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="MyDoubleSpinBox" name="doubleSpinBox_multiple">
       <property name="decimals">
        <number>6</number>
       </property>
       <property name="minimum">
        <double>0.000001000000000</double>
       </property>
       <property name="maximum">
        <double>1.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.000001000000000</double>
       </property>
       <property name="stepType">
        <enum>QAbstractSpinBox::DefaultStepType</enum>
       </property>
       <property name="value">
        <double>1.000000000000000</double>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_5">
       <property name="maximumSize">
        <size>
         <width>30</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>+</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDoubleSpinBox" name="doubleSpinBox_increment">
       <property name="decimals">
        <number>6</number>
       </property>
       <property name="minimum">
        <double>-999999.999998999992386</double>
       </property>
       <property name="maximum">
        <double>999999.999998999992386</double>
       </property>
       <property name="singleStep">
        <double>0.000001000000000</double>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="2" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QLabel" name="label_2">
       <property name="minimumSize">
        <size>
         <width>75</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>75</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>字节顺序</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="comboBox_order">
       <item>
        <property name="text">
         <string>大端模式</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>小端模式</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>大端模式字节交换</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>小端模式字节交换</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </item>
   <item row="4" column="0">
    <widget class="QCheckBox" name="checkBox_event">
     <property name="text">
      <string>配置事件</string>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="8" column="0">
    <widget class="QWidget" name="widget_2" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
   </item>
   <item row="6" column="0">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QFrame" name="frame_2">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="frameShape">
        <enum>QFrame::HLine</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="5" column="0">
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="1" column="0">
        <widget class="QLabel" name="label_type">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>事件类型</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QComboBox" name="event_type">
         <item>
          <property name="text">
           <string>信息</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>告警</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>故障</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="event_name">
         <property name="maxLength">
          <number>64</number>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_condition">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>触发条件</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLabel" name="label_value">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <property name="text">
          <string>寄存器值（换算后）</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_name">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>事件名称</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_8">
       <item>
        <widget class="QCheckBox" name="FlagCondition1">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
         <property name="tristate">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_6">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>范围触发:  </string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="event_range_mode">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="currentIndex">
          <number>0</number>
         </property>
         <property name="maxVisibleItems">
          <number>2</number>
         </property>
         <item>
          <property name="text">
           <string>非必要</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>必要</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QWidget" name="widget" native="true">
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QComboBox" name="event_num_flag">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <item>
             <property name="text">
              <string>介于</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>不介于</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="register_value_min">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="decimals">
             <number>6</number>
            </property>
            <property name="minimum">
             <double>-18446744073709551616.000000000000000</double>
            </property>
            <property name="value">
             <double>0.000000000000000</double>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_10">
            <property name="maximumSize">
             <size>
              <width>30</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>-</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="register_value_max">
            <property name="decimals">
             <number>6</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QFrame" name="frame">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>3</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::HLine</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_9">
       <item>
        <widget class="QCheckBox" name="FlagCondition2">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_7">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>变化阈值触发:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="event_change_mode">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="currentIndex">
          <number>0</number>
         </property>
         <property name="maxVisibleItems">
          <number>2</number>
         </property>
         <item>
          <property name="text">
           <string>非必要</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>必要</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_10">
       <item>
        <widget class="QDoubleSpinBox" name="register_value_change">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>140</width>
           <height>20</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>500</width>
           <height>20</height>
          </size>
         </property>
         <property name="decimals">
          <number>6</number>
         </property>
         <property name="minimum">
          <double>-18446744073709551616.000000000000000</double>
         </property>
         <property name="value">
          <double>0.000000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item row="7" column="0">
    <widget class="QLabel" name="label_9">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>12</height>
      </size>
     </property>
     <property name="styleSheet">
      <string notr="true">color: red;</string>
     </property>
     <property name="lineWidth">
      <number>2</number>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;若使用多个条件，事件是否触发取决于条件之间的&amp;quot;与&amp;quot;&amp;quot;或&amp;quot;关系。非必要为&amp;quot;或关系&amp;quot;，必要为&amp;quot;与&amp;quot;关系. &lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MyDoubleSpinBox</class>
   <extends>QDoubleSpinBox</extends>
   <header location="global">mydoublespinbox.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
